src/primitive/primitive2.rs

```diff
@@ -11,7 +11,7 @@ use crate::{Aabb2, Line2, Ray2};
 /// to use many different primitives in algorithms.
 #[derive(Debug, Clone, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize), serde(crate="serde_crate"))]
-pub enum Primitive2<S> {
+pub enum Primitive2<S:cgmath::BaseNum> {
```

Comment:
should this be `create::BaseNum`?

Reply:
In the primitive.rs file `Primitive3` uses `cgmath::BaseFloat`
```rust
use cgmath::{BaseFloat, Point3, Vector3};
....
pub enum Primitive3<S>
where
    S: BaseFloat,
{

```